### PR TITLE
Feature 1403 python components table

### LINKED ISSUE #1403: Create comprehensive Python package documentation
This task came about from a request from EMC to provide a complete list of Python packages needed for METcalcpy, METplotpy, and METplus wrappers.  Initially created in a Google Spreadsheet, we realize it would be better to provide this information in our existing Sphinx environment.

## Describe the Task ##
Lisa: In the [METplus User's Guide Overview](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html), between section "METplus Wrappers Components" (currently 1.4) and  "METplus Release Notes" (currently 1.5), create a new section called "METplus  Components Python Requirements" (which will become section 1.5).  Copy the information into the spreadsheet in the same format as the [METplus Database of Statistics](https://metplus.readthedocs.io/en/develop/Users_Guide/statistics_list.html).
DONE

Lisa or Julie: Update section "8.4.4.2. Add Sphinx Documentation File" of the [documentation](https://metplus.readthedocs.io/en/develop/Contributors_Guide/add_use_case.html#document-new-use-case) to change the link to the spreadsheet in the text below:

> Update the list of External Dependencies (if applicable) to include any required Python packages. Update the [METplus Components Python Requirements](https://docs.google.com/spreadsheets/d/1Lf-yxZBXTTnBaqCOWUk-jdP9RpaLfil_s-KKH29CkKU/edit#gid=0) spreadsheet. If the package is already listed in the spreadsheet, add a link to the documentation page for this new use case, following the format in the spreadsheet. If the package is not already listed, update the spreadsheet to include the name of the required package, the version, the METplus component (e.g. METplus wrappers, METcalcpy, METplotpy), the source, a brief description, and a link to this new use case that uses this new Python package.

to point to the new section in the documentation.
DONE

Julie: Update the issue templates and the pull request templates to indicate that if there are new Python packages being added for that pull request, check to ensure they are listed in the table referenced above.
DONE

### Time Estimate ###
1 to 3 days of work

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
??

### Funding Source ###
Spread time evenly across 3 accounts:
2702691 
2792541 
2771021

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by jprestop**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>

- Reviewed this [table](https://metplus.readthedocs.io/en/feature_1403_python_components_table/Users_Guide/overview.html#metplus-components-python-requirements) (already merged into develop with a previous pull request, however, there is one new entry):
- Reviewed the Issue and Pull Request templates for formatting.
- Review the part of [Section 8.4.4.2](https://metplus.readthedocs.io/en/feature_1403_python_components_table/Contributors_Guide/add_use_case.html#add-sphinx-documentation-file) that refers to the METplus Components Python Requirements table.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Review the files changed for content, grammar, and formatting (see links above).

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[ASAP]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
 - [ ] Add/update log messages for easier debugging.
 - [ ] Add/update unit tests.
 - [ ] Add/update documentation.
+- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
```
Comment (georgemccabe) [inline .github/ISSUE_TEMPLATE/bug_report.md:68]: I think it is probably unlikely that a bugfix would warrant updating the table, but it doesn't hurt to have it in here.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ If **yes**, describe the new output and/or changes to the existing output:</br>
 
 ## Pull Request Checklist ##
 See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
+- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
```
Comment (georgemccabe) [inline .github/pull_request_template.md:18]: I don't know that this is a step that is needed as part of creating a pull request. This should have been done for completing the issue, so it should be covered there. Again, it doesn't hurt to include it.

File: docs/Contributors_Guide/add_use_case.rst [hunk 3]
```diff
@@ -211,14 +211,14 @@ use case OR category directory for a model_applications use case
 
     * Update the list of External Dependencies (if applicable) to include any
       required Python packages.  Update the
-      `METplus  Components Python Requirements <https://docs.google.com/spreadsheets/d/1Lf-yxZBXTTnBaqCOWUk-jdP9RpaLfil_s-KKH29CkKU/edit#gid=0>`_
-      spreadsheet.  If the package is already listed in the spreadsheet, add
+      `METplus Components Python Requirements <https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements>`_
```
Comment (georgemccabe) [inline docs/Contributors_Guide/add_use_case.rst]: Shouldn't this be a relative path instead of always using develop?

**Review by georgemccabe — CHANGES_REQUESTED** [review]
The changes look mostly good. I added a few in-line comments. The relative path of the link is the more important item of the 3.

File: docs/Contributors_Guide/add_use_case.rst (hunk 3 above)
Comment (jprestop) [reply docs/Contributors_Guide/add_use_case.rst]: @georgemccabe Great suggestion (via slack) to make it a label.  Thanks!  It is now a label.  Please take a look at the Files Changed and the [documentation](https://metplus.readthedocs.io/en/feature_1403_python_components_table/Contributors_Guide/add_use_case.html#add-sphinx-documentation-file) itself.

**Review by georgemccabe — APPROVED** [review]
Looks good after the new changes were added. I approve.

**Pull request merged**